### ISSUE TITLE
How did you limit the answer generation space to just the options you provided?

### ISSUE THREAD

**Issue by Tony363** [lupantech/ScienceQA#12]
Hello,

I am trying to finetune MiniGPT4 on a Student engagement dataset. It labels are image ids and captions about the image. I managed to get it to perform okay on the dataset. However, It would hallucinate or give answer outside of what I would wish for it to answer as MiniGPT-4 is used for Freeform open-ended VQA. 

How did you limit the answer generation space to just the options you provided? The paper mentioned about using a linear classifier at the end to limit the output tokens to just those answer options?

Tony,

**Comment by lupantech** [comment]
Hi Tony,

Thanks for reaching out and expressing interest in our ScienceQA project.

For the zero-shot and few-shot settings, we choose the option that is most similar to the generation from the baseline models like GPT-3.

In the fine-tuning settings, we adopt a method from existing VQA models that builds a linear classifier to predict the class from the most frequent answer candidates found in the training data. Alternatively, we can reframe the problem as a freeform open-ended task, and then select the most similar option as the final prediction answer.

I hope this clarifies your query. Please feel free to reach out if you have any more questions.

Best,
Pan
